tokenbucket.go

```diff
@@ -73,13 +73,13 @@ func NewTokenBucket(capacity int64, refillRate time.Duration, locker DistLocker,
 	}
 }
 
-// TakeMinMax takes up to `max` tokens from the bucket, when it can take at least `minimum`.
+// TakeMinMax takes between minTokens and maxTokens tokens from the bucket, depending on availability.
 //
-// It returns a zero duration and a nil error if the bucket has sufficient number of tokens.
+// It returns the number of tokens actually taken, zero duration, and nil error if at least minTokens are available.
 //
-// It returns ErrLimitExhausted if the number of available tokens is less than the minimum requested.
-// In this case the returned  duration is the amount of time to wait to retry the request.
-func (t *TokenBucket) TakeMinMax(ctx context.Context, minimum, tokens int64) (int64, time.Duration, error) {
+// If fewer than minTokens are available, it returns 0 taken, ErrLimitExhausted, and a duration indicating how long to wait before retrying.
```

Comment:
`0 taken, ErrLimitExhausted, and a duration` - the actual order is different: tokens taken, duration, error